x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -67,300 +128,357 @@ export const DeploymentSetup: FC<DeploymentSetupProps> = ({
-  } = getNewJobLimits();
-
-  const numOfAllocation = config.numOfAllocations;
-  const threadsPerAllocations = config.threadsPerAllocations;
+  const deploymentIdUpdated = useRef(false);
 
   const defaultDeploymentId = useMemo(() => {
     return config.deploymentId;
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const threadsPerAllocationsOptions = useMemo(
-    () =>
-      new Array(THREADS_MAX_EXPONENT)
-        .fill(null)
-        .map((v, i) => Math.pow(2, i))
-        .filter(maxSingleMlNodeProcessors ? (v) => v <= maxSingleMlNodeProcessors : (v) => true)
-        .map((value) => {
-          const id = value.toString();
-
-          return {
-            id,
-            label: id,
-            value,
-            'data-test-subj': `mlModelsStartDeploymentModalThreadsPerAllocation_${id}`,
-          };
-        }),
-    [maxSingleMlNodeProcessors]
-  );
-
-  const disableThreadingControls = config.priority === 'low';
+  const customTicks = Object.values(vCpuLevelMap).map((v) => {
+    return {
+      label: v.label,
+      value: v.value,
+    };
+  });
+
+  const customColorsLevels = [
+    {
+      min: 0.5,
+      max: 1.1,
+      color: vCpuLevelMap.low.color,
+    },
+    {
+      min: 1.1,
+      max: 1.9,
+      color: vCpuLevelMap.medium.color,
+    },
+    {
+      min: 1.9,
+      max: 2.5,
+      color: vCpuLevelMap.high.color,
+    },
+  ];
+
+  const optimizedOptions = [
+    {
+      id: 'optimizedForIngest',
+      value: 'optimizedForIngest' as const,
+      label: (
+        <EuiText size={'s'}>
+          <strong>
+            <FormattedMessage
+              id="xpack.ml.trainedModels.modelsList.startDeployment.optimizedForIngestLabel"
+              defaultMessage="Ingest"
+            />
+          </strong>
+        </EuiText>
+      ),
+      description: (
+        <FormattedMessage
+          id="xpack.ml.trainedModels.modelsList.startDeployment.optimizedForIngestDescription"
+          defaultMessage="Optimized for higher throughput during ingest"
+        />
+      ),
+      'data-test-subj': `mlModelsStartDeploymentModalOptimizedForIngest`,
+    },
+    {
+      id: 'optimizedForSearch',
+      value: 'optimizedForSearch' as const,
+      label: (
+        <FormattedMessage
+          id="xpack.ml.trainedModels.modelsList.startDeployment.optimizedForSearchLabel"
+          defaultMessage="Search"
+        />
+      ),
+      description: (
+        <FormattedMessage
+          id="xpack.ml.trainedModels.modelsList.startDeployment.optimizedForSearchDescription"
+          defaultMessage="Optimized for lower latency during search"
+        />
+      ),
+      'data-test-subj': `mlModelsStartDeploymentModalOptimizedForSearch`,
+    },
+  ];
+
+  const helperText = useMemo<string>(() => {
+    const vcpuRange = deploymentParamsMapper.getVCPURange(config.vCPUUsage);
+
+    if (config.adaptiveResources) {
+      switch (config.vCPUUsage) {
```

Comment:
I see this messages referring to the Cloud console when running locally (dev mode). Is this to be expected?